vmware_aria_operations_integration_sdk/project.py

```diff
@@ -104,16 +103,43 @@ def get_memory_limit(self) -> int:
         return memory_limit  # type: ignore
 
     @classmethod
-    def extract(cls, json_connection: Dict) -> Connection:
+    def extract(cls, path: str, json_connection: Dict) -> Connection:
         name = json_connection[CONNECTIONS_CONFIG_CONNECTION_NAME_KEY]
         identifiers = json_connection[CONNECTIONS_CONFIG_CONNECTION_IDENTIFIERS_KEY]
         credential = json_connection[CONNECTIONS_CONFIG_CONNECTION_CREDENTIAL_KEY]
         certificates = json_connection.get(
             CONNECTIONS_CONFIG_CONNECTION_CERTIFICATES_KEY, None
         )
-        hostname = json_connection.get(CONNECTIONS_CONFIG_SUITE_API_HOSTNAME_KEY, None)
-        username = json_connection.get(CONNECTIONS_CONFIG_SUITE_API_USERNAME_KEY, None)
-        password = json_connection.get(CONNECTIONS_CONFIG_SUITE_API_PASSWORD_KEY, None)
+
+        default_hostname = get_config_value(
+            CONNECTIONS_CONFIG_SUITE_API_HOSTNAME_KEY,
+            "hostname",
```

Comment:
No, because the schema defines these as non-null (and non-empty), and validation will fail if we don't have a value of some sort. Alternatively, we could modify the schema. I didn't do that to try to keep schema changes to a minimum, so that if the schema is updated there is less work required to merge the changes.

Reply:
?

Reply:
I don't think we should modify the schema.
1. We don't control it - it's created by the platform team
2. When it's updated we'll have to merge any changes we've made with changes the platform team has made. The more updates we make the more error-prone this process becomes.
3. We want the behavior of the test tool to exactly match the behavior of the platform. The best way we have to ensure that is to follow their schema as closely as possible.